Snowflake.Data/Core/SFResultSet.cs

```diff
@@ -183,6 +183,16 @@ internal override async Task<bool> NextResultAsync(CancellationToken cancellatio
             return await Task.FromResult(false);
         }
 
+        internal override bool HasRows()
+        {
+            if (isClosed)
+            {
+                throw new SnowflakeDbException(SFError.DATA_READER_ALREADY_CLOSED);

```

Comment:
do we need to throw an exception or we can just return false?

Reply:
I was based on Next(), but you're right. Changed to return false.